dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/acl/TrackerProgramService.java

```diff
@@ -52,9 +58,39 @@ public class TrackerProgramService {
   @Nonnull private final ProgramService programService;
   @Nonnull private final AclService aclService;
 
+  /**
+   * Returns the tracker program associated with the provided UID if it exists and is accessible to
+   * the current user.
+   */
+  @Transactional(readOnly = true)
+  public @Nonnull Program getTrackerProgram(@CheckForNull UID programUid)
+      throws BadRequestException, ForbiddenException {
+    if (programUid == null) {
```

Comment:
I would personally give this responsibility to the caller and annotate `programUid` as `@Nonnull`